### PR TITLE
Partial

### PR DESCRIPTION
**Pull request by gerardofn**

### CODE REVIEW

File: plugin/virustotal/vt_ida/disassembler.py [hunk 1]
```diff
@@ -15,12 +15,119 @@
...
+
+  @staticmethod
+  def is_head(ea):
+    flags = idc.get_full_flags(ea)
+    return idc.is_head(flags)
+
+  @staticmethod
+  def is_imports(segment):
+    if segment in ('.idata', '.rdata', '.edata'):
+      # Check: .dynstr, '.dynsym', __IMPORT
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/disassembler.py:44]: Please go over all the code, and everywhere there is still work to be done, add

TODO(gerardofn): lo q sea

File: README.md [hunk 2]
```diff
@@ -34,6 +42,9 @@ Another option is to look for similar strings. To search for similar ones, open
 
 These actions will launch a new instance of your default web browser, showing all the matches found in VTGrep. Remember that your default web browser must be logged into your VirusTotal Enterprise account in order to see the results.
 
-Check IDA Pro's output window for any message that may need your attention.
-
 **Note**: This version supports **Intel 32/64 bits** and **ARM** processor architectures when searching for similar code. Probably more architectures are supported but it hasn't been tested yet.
+
+### Navigate through VTPanel
+TBA
```
Comment (winden-g) [inline README.md:48]: replace TBA with TODO(....

File: plugin/virustotal/config.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
 # VT-IDA plugin global configuration file
 
-API_KEY = ''  # Optional
+API_KEY = ''  # Required for fetching VT Intelligence info, can be public or private API KEY
```
Comment (winden-g) [inline plugin/virustotal/config.py:4]: Is it still optional?

File: plugin/virustotal/vt_ida/check_sample.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    """Return True if the file is not available at VirusTotal."""
+
+    user_agent = 'IDA Pro VT Plugin checkhash - v'
+    user_agent += defaults.VT_IDA_PLUGIN_VERSION
+    headers = {
+        'User-Agent': user_agent,
+        'Accept': 'application/json'
+    }
+
+    url = 'https://www.virustotal.com/ui/files/%s' % self.file_hash
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/check_sample.py:76]: why ui and not api?

File: plugin/virustotal/vt_ida/check_sample.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+          'User-Agent': user_agent,
+          'x-apikey': config.API_KEY
+      }
+
+    norm_path = os.path.normpath(self.input_file)
+    file_path, file_name = os.path.split(norm_path)
+
+    if os.path.isfile(self.input_file):
+      logging.info('[VT Plugin] Uploading input file to VirusTotal.')
+      url = 'https://www.virustotal.com/ui/files'
```
Comment (winden-g) [inline plugin/virustotal/vt_ida/check_sample.py:111]: Again, why UI and not API?

**Review by winden-g — CHANGES_REQUESTED** [review]

File: plugin/virustotal/config.py (hunk 3 above)
Comment (gerardofn) [reply plugin/virustotal/config.py:4]: Yes, when the API KEY is not defined it just provides the same functionality than v0.10 (vtgrep integration)

**Pull request merged**